### PR TITLE
Leaflet maps on Strava as input

### LINKED ISSUE #20: Support for Strava Route Builder
I am interested in adding support for the Strava Route Builder to MapSwitcher, as an input. This is the page with a URL like "https://www.strava.com/routes/26021009/edit" (though that probably won't work for you, because it's my route :P).

I've skimmed some of the techniques you use in dataExtractor.js, like URL fragments, permalinks, etc. The ones I've seen don't seem to apply to this particular page. This page has an embedded Leaflet map viewer. I'm curious if you have any leads on how to deal with those. (It seems possible I could look at tile URLs & positions and reverse-engineer the view from that...)

P.S. MapSwitcher is fantastic! It is the map-switcher of my dreams. Thank you for sharing it!

### PR DESCRIPTION
**Pull request by joshuahhh**
Resolves #20, with the added benefit that it makes _all_ Leaflet maps on Strava work as inputs, including ride maps. (Perhaps this could be made into an all-host input, so Leaflet maps anywhere can work as inputs?)

Please let me know what you think! Thanks.

### CODE REVIEW

**Comment by joshuahhh** [comment]
Hi @david-r-edgar! Is it possible for you to take a look at this? Thanks!

**Comment by david-r-edgar** [comment]
> Hi @david-r-edgar! Is it possible for you to take a look at this? Thanks!

Ah, sorry, missed the notification before. Will have a look, thanks for the PR!

File: src/dataExtractor.js [hunk 1]
```diff
@@ -887,16 +887,50 @@ extractors.push({
...
+      function tileMethod () {
+        const container = document.querySelector('.leaflet-container')
+        if (!container) { return false }
+        const containerRect = container.getBoundingClientRect()
+
+        const tile = container.querySelector('.leaflet-tile')
+        if (!tile) { return false }
+        const tileRect = tile.getBoundingClientRect()
+
+        const re = /\/([0-9]+)\/([0-9]+)\/([0-9]+)@2x.png/
```
Comment (david-r-edgar) [inline src/dataExtractor.js]: I tried it, and it failed to match here.
But when I removed `@2x` it did succeed (for both rides and routes).
Do you have any idea why I don't see this in the tile URL and you, I assume, do?

**Comment by joshuahhh** [comment]
Ooooo I wonder if that's a retina thing! (I'm using a MacBook which has a
2x pixel density screen.) I suppose we can just remove that part. I'll
revise the PR in the next day to reflect that. Thanks for testing!

**Comment by joshuahhh** [comment]
I was able to test this on 2x and 1x resolution mode, so I bet it works! Would be great to get a double-check on your end, though. (Btw: The route planner interface recently changed and it's now based on a canvas. I don't know how to deal with that yet. But this still works for all the tile-based maps around Strava.) Thanks!

**Review by david-r-edgar — APPROVED** [review]
All looks good. Tested locally against various rides and routes.

**Comment by david-r-edgar** [comment]
Thanks for the fix! Seems to work well.

**Pull request merged**